### ISSUE TITLE
DaemonSets will conflict with hostPorts on nodes it shouldn't apply to

### ISSUE THREAD

**Issue by lmickh** [kubernetes/kubernetes#43104]
Bug report

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.4", GitCommit:"7243c69eb523aa4377bce883e7c0dd76b84709a1", GitTreeState:"clean", BuildDate:"2017-03-08T02:50:34Z", GoVersion:"go1.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3+coreos.0", GitCommit:"8fc95b64d0fe1608d0f6c788eaad2c004f31e7b7", GitTreeState:"clean", BuildDate:"2017-02-15T19:52:15Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: Azure
- **OS** (e.g. from /etc/os-release): CoreOS 1298.5.0
- **Kernel** (e.g. `uname -a`): 4.9.9-coreos-r1
- **Install tools**:  Manual

**What happened**:

Attempted to run a daemonset that includes a hostPort and is using a nodeSelector.  The controller will fail to place to the pod on nodes that do not match the nodeSelector if there is a port conflict with one of the hostPorts.

**What you expected to happen**:

The controller should not try to place a pod on a node that does not match the nodeSelector within a daemonset.

**How to reproduce it** (as minimally and precisely as possible):

Assuming the master node has the API running on 443, create a daemonset with a hostPort 443 and a nodeSelector (ex. role=edge).  Use `kubectl apply` to install the daemonset.  The controller should indicate it is trying to place the pod with an event message similar to this:
```
reason: 'FailedPlacement' failed to place pod on "master-1": host port conflict
```

Remove the conflicting hostPort and the daemonset will be applied without attempting to be placed on the node that does not match the nodeSelector.

**Comment by lukaszo** [comment]
It looks like it's fixed in 1.6(master) I couldn't reproduce it. Does it somehow block you? It seems to be just an UX issue.

**Comment by k82cn** [comment]
daemonset is not care about nodeSelector; it dispatch one pod per node by controller. But in controller, it re-used scheduler's predicts for port conflict.

**Comment by lukaszo** [comment]
@k82cn Daemonsets are using nodeSelector. Please see the documentation for more information https://kubernetes.io/docs/admin/daemons/#running-pods-on-only-some-nodes

**Comment by k82cn** [comment]
@lukaszo , got that, misunderstand :).

**Comment by k82cn** [comment]
@lukaszo , got that, misunderstand :).

**Comment by lmickh** [comment]
@lukaszo It is not a blocker for myself.  Mainly log noise at the moment.  Could imagine on a larger cluster it adds up under the right conditions.  Feel free to close it if it is addressed in 1.6.  I haven't had a chance to test it.

**Comment by lukaszo** [comment]
@lmickh I don't have rights to close issues.
